packages/kit/types/config.d.ts

```diff
@@ -1,38 +1,85 @@
 import { UserConfig as ViteConfig } from 'vite';
 import { RecursiveRequired } from './helper';
-import { Logger, TrailingSlash } from './internal';
+import { HttpMethod, Logger, RouteSegment, TrailingSlash } from './internal';
 
-export interface AdapterUtils {
+export interface RouteDefinition {
+	type: 'page' | 'endpoint';
+	pattern: RegExp;
+	segments: RouteSegment[];
+	methods: HttpMethod[];
+}
+
+export interface AdapterEntryConfig {
+	id: string;
+	filter: (route: RouteDefinition) => boolean;
+	complete: (entry: AdapterEntry) => void;
+}
+
+export interface AdapterEntry {
+	generateManifest: (opts: { relativePath: string; format?: 'esm' | 'cjs' }) => string;
+}
+
+export interface Builder {
 	log: Logger;
 	rimraf(dir: string): void;
 	mkdirp(dir: string): void;
+
+	/**
+	 * Create entry points that map to individual functions
+	 * @param fn TODO explain this
```

Comment:
this TODO seems easy enough to fix now

Reply:
done, though it's a very bare-bones description — I don't think we can fully explain this method without some documentation and examples